gtars/src/uniwig/counting.rs

```diff
@@ -0,0 +1,244 @@
+/// This function is a more direct port of smoothFixedStartEndBW from uniwig written in CPP.
+/// It allows the user to accumulate reads of either starts or ends.
+/// Counts occur between a start coordinate (cutSite) and an end site (endSite) where the endsite is determined based on
+/// the level of smoothing.
+/// counts are reported over a stepsize (with a default of stepsize = 1).
+/// Unlike the original function, it does not write to disk in chunks. it simply returns a vector of accumulated reads.
+#[allow(unused_variables)]
+pub fn start_end_counts(
+    starts_vector: &Vec<(i32, i32)>,
+    chrom_size: i32,
+    smoothsize: i32,
+    stepsize: i32,
+) -> (Vec<u32>, Vec<i32>) {
+    //let vin_iter = starts_vector.iter();
+
+    let mut v_coordinate_positions: Vec<i32> = Vec::new(); // these are the final coordinates after any adjustments
+    let mut v_coord_counts: Vec<u32> = Vec::new(); // u8 stores 0:255 This may be insufficient. u16 max is 65535
+
+    let mut coordinate_position = 1;
+
+    let mut count: i32 = 0;
+
+    let mut coordinate_value: (i32, i32);
+    let mut prev_coordinate_value = 0;
+
+    let mut adjusted_start_site: (i32, i32);
+    let mut current_end_site: (i32, i32);
+
+    let mut collected_end_sites: Vec<(i32, i32)> = Vec::new();
+
+    adjusted_start_site = starts_vector[0].clone(); // get first coordinate position
+
+    adjusted_start_site.0 = adjusted_start_site.0 - smoothsize;
+
+    current_end_site = adjusted_start_site;
+    current_end_site.0 = adjusted_start_site.0 + 1 + smoothsize * 2;
+
+    if adjusted_start_site.0 < 1 {
+        adjusted_start_site.0 = 1;
+    }
+
+    while coordinate_position < adjusted_start_site.0 {
+        // Just skip until we reach the initial adjusted start position
+        // Note that this function will not return 0s at locations before the initial start site
+        coordinate_position = coordinate_position + stepsize;
+    }
+
+    for (index, coord) in starts_vector.iter().enumerate().skip(0) {
```

Comment:
Doesnt the `.skip(0)` here just say, "Skip nothing" which is the same as not including it at all?